src/Transport/Config/DefaultConfigurationValues.cs

```diff
@@ -1,13 +1,21 @@
 ﻿namespace NServiceBus.AzureServiceBus
 {
     using System;
+    using System.Text.RegularExpressions;
+    using Addressing;
     using Microsoft.ServiceBus;
     using Microsoft.ServiceBus.Messaging;
     using Topology.MetaModel;
     using Settings;
 
     class DefaultConfigurationValues
     {
+        // Entity segments can contain only letters, numbers, periods (.), hyphens (-), and underscores (-), paths can contain slashes (/)
+        static Regex regex = new Regex(@"^[^\/][0-9A-Za-z_\.\-\/]+[^\/]$");
+
+        // Except for subscriptions and rules, these cannot contain slashes (/)
+        static Regex subscriptionAndRuleNameRegex = new Regex(@"^[0-9A-Za-z_\.\-]+$");
+
```

Comment:
@yvesgoeleven I thought about it and really can't come up with a reason to do so.
I'd rather have it documented (or provide a sample) then expose in the code and later regret that we cannot retire it if needed. I've [captured](https://github.com/Particular/NServiceBus.AzureServiceBus/issues/252#issuecomment-227333102) what was validate/sanitization for V6 (and validation for V7 we're currently using) in the issue.